spec/system/candidate_interface/candidate_new_user_with_course_params_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+RSpec.describe 'A new candidate arriving from Find with a course and provider code' do
+  include FindAPIHelper
+  include ActiveJob::TestHelper
+
+  scenario 'retaining their course selection through the sign up process' do
+    given_the_pilot_is_open
+    and_confirm_course_choice_from_find_is_activated
+    and_the_course_i_selected_only_has_one_site
+
+    when_i_arrive_from_find_to_a_course_that_is_open_on_apply
+    and_i_click_apply_on_apply
+    then_the_url_should_contain_the_course_code_and_provider_code_param
+    when_i_fill_in_the_eligiblity_form_with_yes
+    then_the_url_should_contain_the_course_code_and_provider_code_param
```

Comment:
Perhaps we drop the tests for URLs? They are an implementation detail - generally in these tests we shouldn't care about _how_ something happens, just that it does.